src/SurveyService.Validation/Option/CreateOptionRequestValidator.cs

```diff
@@ -7,20 +7,24 @@ namespace LT.DigitalOffice.SurveyService.Validation.Option;
 public class CreateOptionRequestValidator : AbstractValidator<CreateOptionRequest>, ICreateOptionRequestValidator
 {
   private readonly IQuestionRepository _questionRepository;
+
   public CreateOptionRequestValidator(
     IQuestionRepository questionRepository)
   {
     _questionRepository = questionRepository;
 
     RuleFor(option => option.Content)
-      .Cascade(CascadeMode.Stop)
-      .NotEmpty().WithMessage("Content can't be empty.")
-      .MinimumLength(1).WithMessage("Content is too short.")
-      .MaximumLength(300).WithMessage("Content is too long.");
+      .MaximumLength(300)
+      .WithMessage("Content is too long.");
 
-    RuleFor(option => option)
-      .MustAsync(async (o, _) =>
-        (await _questionRepository.GetAsync(o.QuestionId))?.HasCustomOptions == o.IsCustom)
-      .WithMessage("Wrong IsCustom value.");
+    When(option => option.IsCustom, () =>
+    {
+      RuleFor(option => option.QuestionId)
+      .Cascade(CascadeMode.Stop)
+      .MustAsync(async (option, _) => await _questionRepository.GetAsync(option) is not null)
+      .WithMessage("Can't create option for non-existent question.")
+      .MustAsync(async (option, _) => await _questionRepository.GetAsync(option).HasCustomOptions)
+      .WithMessage("This question can't have custom options.");
+    });
```

Comment:
Can an option with IsCustom = true come from CreateQuestionRequestValidator at all? If not, then is it worth doing a check with HasCustomOptions? So far, I'm only checking for custom options.